panther_hardware_interfaces/include/panther_hardware_interfaces/roboteq_driver.hpp

```diff
@@ -0,0 +1,197 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef PANTHER_HARDWARE_INTERFACES_ROBOTEQ_DRIVER_HPP_
+#define PANTHER_HARDWARE_INTERFACES_ROBOTEQ_DRIVER_HPP_
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <string>
+
+#include <lely/coapp/fiber_driver.hpp>
+
+namespace panther_hardware_interfaces
+{
+
+struct RoboteqMotorState
+{
+  std::int32_t pos;
+  std::int32_t vel;
+  std::int32_t current;
+};
+
+struct RoboteqDriverFeedback
+{
+  RoboteqMotorState motor_1;
+  RoboteqMotorState motor_2;
+
+  std::uint8_t fault_flags;
+  std::uint8_t script_flags;
+  std::uint8_t runtime_stat_flag_motor_1;
+  std::uint8_t runtime_stat_flag_motor_2;
+
+  timespec timestamp;
+};
+
+// todo: heartbeat timeout (on hold - waiting for decision on changing to PDO)
+/**
+ * @brief Implementation of FiberDriver for Roboteq drivers
+ */
+class RoboteqDriver : public lely::canopen::FiberDriver
+{
+public:
+  using FiberDriver::FiberDriver;
+
+  RoboteqDriver(
+    const std::shared_ptr<lely::ev::Executor> & exec,
+    const std::shared_ptr<lely::canopen::AsyncMaster> & master, const std::uint8_t id,
+    const std::chrono::milliseconds & sdo_operation_timeout_ms);
+
+  /**
+   * @brief Trigger boot operations
+   */
+  bool Boot();
+
+  /**
+   * @brief Waits until the booting procedure finishes
+   *
+   * @exception std::runtime_error if boot fails
+   */
+  bool WaitForBoot();
+
+  bool IsBooted() const { return booted_.load(); }
+
+  bool IsCANError() const { return can_error_.load(); }
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadTemperature();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::uint16_t ReadVoltage();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadBatAmps1();
+
+  /**
+   * @exception std::runtime_error if operation fails
+   */
+  std::int16_t ReadBatAmps2();
+
+  /**
+   * @brief Reads all the PDO data returned from Roboteq (motors feedback, error flags) and saves
+   * current timestamp
+   */
+  RoboteqDriverFeedback ReadRoboteqDriverFeedback();
+
+  /**
+   * @param cmd command value in the range [-1000, 1000]
+   * @exception std::runtime_error if operation fails
+   */
+  void SendRoboteqCmdChannel1(const std::int32_t cmd);
+
+  /**
+   * @param cmd command value in the range [-1000, 1000]
+   * @exception std::runtime_error if operation fails
+   */
+  void SendRoboteqCmdChannel2(const std::int32_t cmd);
```

Comment:
Same here. Could we provide a brief description of what Chanel1 and Chanel2 represent?